core/include/drivers/gpio.h

```diff
@@ -133,6 +135,39 @@ static inline void gpio_put(struct gpio *gpio)
 		gpio->chip->ops->put(gpio->chip, gpio);
 }
 
+#define GPIO_FLAGS_BIT_DIR_SET		BIT(0)
+#define GPIO_FLAGS_BIT_DIR_OUT		BIT(1)
+#define GPIO_FLAGS_BIT_DIR_VAL		BIT(2)
+
+/**
+ * enum gpio_flags - Optional flags that used to configure direction and output
+ *                   value. These values cannot be OR'd.
+ *
+ * @GPIOD_ASIS:		Don't change anything
+ * @GPIO_IN:		Set lines to input mode
+ * @GPIO_OUT_LOW:	Set lines to output and drive them low
+ * @GPIO_OUT_HIGH:	Set lines to output and drive them high
+
+ */
+enum gpio_flags {
+	GPIO_ASIS	= 0,
+	GPIO_IN		= GPIO_FLAGS_BIT_DIR_SET,
+	GPIO_OUT_LOW	= GPIO_FLAGS_BIT_DIR_SET | GPIO_FLAGS_BIT_DIR_OUT,
+	GPIO_OUT_HIGH	= GPIO_FLAGS_BIT_DIR_SET | GPIO_FLAGS_BIT_DIR_OUT |
+			  GPIO_FLAGS_BIT_DIR_VAL,
+};
+
+/**
+ * gpio_configure() - Configure a GPIO controller
+ *
+ * @gpio: GPIO pin reference upon success
+ * @flags: requester flags of GPIO
```

Comment:
s/flags/mode/
Or maybe the a more consistent change would be to rename the argument from `mode` to `flags`.

Ditto in `gpio_dt_cfg_by_index()` inline description comment.